ci-operator/step-registry/upi/install/vsphere/upi-install-vsphere-commands.sh

```diff
@@ -194,15 +241,22 @@ ret="$?"
 set -e
 
 if [ $ret -ne 0 ]; then
-  set +e
-  # Attempt to gather bootstrap logs.
-  echo "$(date -u --rfc-3339=seconds) - Bootstrap failed, attempting to gather bootstrap logs..."
-  gather_console_and_bootstrap
-  sed 's/password: .*/password: REDACTED/' "${installer_dir}/.openshift_install.log" >>"${ARTIFACT_DIR}/.openshift_install.log"
-  echo "$(date -u --rfc-3339=seconds) - Copy log-bundle to artifacts directory..."
-  cp --verbose "${installer_dir}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}"
-  set -e
-  exit "$ret"
+  if [ X"${CCM_TYPE}" == X"External" ]; then
+    echo "Installation failed but 'ccm_type' is External, adding pending steps...."
+    deploy_ccm || exit 1
```

Comment:
Hmm, the deploy_ccm should happen after the 1st install failure, but not after bootstrap phase?

Reply:
It should happen after bootstrap phase, because the ccm is required for bootstrap.